internal/app/cache/cache.go

```diff
@@ -0,0 +1,25 @@
+package cache
+
+import (
+	envoy_api_v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+)
+
+type Cache interface {
+	// Exists returns true if the key exists in the cache, false otherwise.
+	Exists(key string) bool
+
+	// Fetch returns the cached response if it exists.
+	Fetch(key string) (*envoy_api_v2.DiscoveryResponse, error)
+
+	// SetResponse sets the cache response and returns the list of open watches.
+	SetResponse(key string, resp envoy_api_v2.DiscoveryResponse) ([]*envoy_api_v2.DiscoveryRequest, error)
+
+	// ClearWatches discards the existing watches.
+	ClearWatches(key string) error
```

Comment:
@LisaLudique Do we envision ClearWatches to be called in the scenario of TTL expiration?

Reply:
No, for TTL expiration I imagine we'd clear the entire entry for a key, not just the watches field. I thought this field would be relevant when we get a DiscoveryResponse from the control plane - per the sequence diagram and #9, this method is used to discard to watches when a response is set for a given key.

However, I can see this method also being built into SetResponse, so we may not need it as a separate public method.